sagemaker-remote-function/huggingface_text_classification/huggingface.ipynb

```diff
@@ -0,0 +1,371 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5ff54938",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# tokenizer helper function\n",
+    "def tokenize(batch):\n",
+    "    return tokenizer(batch['text'], padding='max_length', truncation=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "90c643cc",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "train_dataset, test_dataset = load_dataset('imdb', split=['train', 'test'])\n",
+    "# for demo, smaller the size of the datasets\n",
+    "test_dataset = test_dataset.shuffle().select(range(5000))\n",
+    "\n",
+    "# tokenize dataset\n",
+    "train_dataset = train_dataset.map(tokenize, batched=True)\n",
+    "test_dataset = test_dataset.map(tokenize, batched=True)\n",
+    "\n",
+    "# set format for pytorch\n",
+    "train_dataset =  train_dataset.rename_column(\"label\", \"labels\")\n",
+    "train_dataset.set_format('torch', columns=['input_ids', 'attention_mask', 'labels'])\n",
+    "test_dataset = test_dataset.rename_column(\"label\", \"labels\")\n",
+    "test_dataset.set_format('torch', columns=['input_ids', 'attention_mask', 'labels'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a2ed32ff",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "train_data_path = os.path.join(s3_root_folder, 'data', 'train')\n",
+    "test_data_path = os.path.join(s3_root_folder, 'data', 'test')\n",
+    "\n",
+    "train_dataset.save_to_disk(train_data_path)\n",
+    "test_dataset.save_to_disk(test_data_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cceaa51b",
+   "metadata": {},
+   "source": [
+    "## Run the training remotely with a GPU instance\n",
+    "\n",
+    "\n",
+    "The following method is used to compute metrics that evaluate the binary classification."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ff2589b8",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from sklearn.metrics import accuracy_score, precision_recall_fscore_support\n",
+    "\n",
+    "def compute_metrics(pred):\n",
+    "    labels = pred.label_ids\n",
+    "    preds = pred.predictions.argmax(-1)\n",
+    "    precision, recall, f1, _ = precision_recall_fscore_support(labels, preds, average=\"binary\")\n",
+    "    acc = accuracy_score(labels, preds)\n",
+    "    return {\"accuracy\": acc, \"f1\": f1, \"precision\": precision, \"recall\": recall}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "39871063",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import s3fs\n",
+    "import json\n",
+    "\n",
+    "from sagemaker.remote_function import remote\n",
+    "\n",
+    "@remote(s3_root_uri=s3_root_folder, keep_alive_period_in_seconds=600)\n",
```

Comment:
If we are depending on config.yaml for any of the config, then we first need to export the config file path before invoking remote function

```
os.environ["SAGEMAKER_USER_CONFIG_OVERRIDE"]="path/to/config/file" (can be os.getcwd())
```